### PR TITLE
Fix packaging of unwanted dependencies

### PR DESCRIPTION
**Pull request by Vlatombe**
* instance-identity scope should be provided
* ssh-credentials 1.18
* fix bundled dependencies that shouldn't be packaged
(https://github.com/jenkinsci/maven-hpi-plugin/pull/140 doesn't cover
all use cases)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -144,6 +144,16 @@
...
       <classifier>tests</classifier>
       <scope>test</scope>
+      <exclusions>
+        <exclusion>
+          <groupId>org.jboss.marshalling</groupId>
+          <artifactId>jboss-marshalling-river</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>org.jboss.marshalling</groupId>
+          <artifactId>jboss-marshalling</artifactId>
```
Comment (jglick) [inline pom.xml]: Hmm, why was this necessary?

File: pom.xml [hunk 2]
```diff
@@ -265,6 +250,13 @@
         <scope>import</scope>
         <type>pom</type>
       </dependency>
+      <dependency>
+        <!-- Avoid it from being bundled -->
```
Comment (jglick) [inline pom.xml]: Ditto, where was this coming from before?

File: pom.xml (hunk 1 above)
Comment (Vlatombe) [reply pom.xml]: Not sure but it seems it was pulled in packaging according to build logd

Comment (Vlatombe) [reply pom.xml]: I tracked it down to the addition of workflow-cps in https://github.com/jenkinsci/kubernetes-plugin/pull/707 which is pulling workflow-support transitively and then altering the scope of following dependencies (from `test` to `compile`)

Comment (Vlatombe) [reply pom.xml]: An occurrence of https://github.com/jenkinsci/maven-hpi-plugin/pull/140#discussion_r322739625 essentially

Comment (Vlatombe) [reply pom.xml]: Trying out a new fix of maven-hpi-plugin

Comment (Vlatombe) [reply pom.xml]: https://github.com/jenkinsci/maven-hpi-plugin/pull/172/

File: pom.xml (hunk 2 above)
Comment (Vlatombe) [reply pom.xml]: Compile scope comes this trail

```
+- org.jenkinsci.plugins:kubernetes-credentials:jar:0.6.2:compile
|  +- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:jar:4.5.10-2.0:compile
|  |  +- org.apache.httpcomponents:httpclient:jar:4.5.10:compile
|  |  |  \- org.apache.httpcomponents:httpcore:jar:4.4.12:compile
|  |  +- org.apache.httpcomponents:httpmime:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:fluent-hc:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:httpclient-cache:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
|  |  |  \- org.apache.httpcomponents:httpcore-nio:jar:4.4.10:compile
|  |  \- org.apache.httpcomponents:httpasyncclient-cache:jar:4.1.4:compile
|  \- org.jenkins-ci.plugins:google-oauth-plugin:jar:0.8:compile
|     +- joda-time:joda-time:jar:2.9.5:compile
|     +- org.jenkins-ci.plugins:oauth-credentials:jar:0.3:compile
|     |  \- net.sourceforge.findbugs:jsr305:jar:1.3.2:compile
```

Comment (Vlatombe) [reply pom.xml]: Trails
```
+- org.jenkinsci.plugins:kubernetes-credentials:jar:0.6.2:compile
|  +- org.jenkins-ci.plugins:apache-httpcomponents-client-4-api:jar:4.5.10-2.0:compile
|  |  +- org.apache.httpcomponents:httpclient:jar:4.5.10:compile
|  |  |  \- org.apache.httpcomponents:httpcore:jar:4.4.12:compile
|  |  +- org.apache.httpcomponents:httpmime:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:fluent-hc:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:httpclient-cache:jar:4.5.10:compile
|  |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.4:compile
|  |  |  \- org.apache.httpcomponents:httpcore-nio:jar:4.4.10:compile
|  |  \- org.apache.httpcomponents:httpasyncclient-cache:jar:4.1.4:compile
|  \- org.jenkins-ci.plugins:google-oauth-plugin:jar:0.8:compile
|     +- joda-time:joda-time:jar:2.9.5:compile
|     +- org.jenkins-ci.plugins:oauth-credentials:jar:0.3:compile
|     |  \- net.sourceforge.findbugs:jsr305:jar:1.3.2:compile
|     +- com.google.http-client:google-http-client:jar:1.24.1:compile
|     |  \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
```

```
+- com.github.spotbugs:spotbugs-annotations:jar:3.1.12:provided (optional) 
|  \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
```

The one from spotbugs-annotations is much shorter, so this is the one that gets picked up during packaging.
